controllers/main.go

```diff
@@ -1595,13 +1595,14 @@ func (controller *MainController) LoginPost(c web.C, r *http.Request) (string, i
 	user, err := helpers.Login(dbMap, email, password)
 	if err != nil {
 		log.Infof(email+" login failed %v, %v", err, remoteIP)
-		session.AddFlash("Invalid Email or Password", "loginError")
```

Comment:
So as you can see I'm using `c.Env["IsInvalidLogin"]` is there a reason why error messages were handled as session flash messages ? with `c.Env["IsInvalidLogin"]` I was able to toggle the error message div (show/hide) but also add/remove class from the input to change the styling when error occurs.
Would like to get your opinion before moving forward with my implementation.

Reply:
Perhaps to allow the error string to be dynamic?  And also it's gorilla, so it's cool.  Something about good for redirects. I would think you can still do the add/remove stuff with the flash message, but I don't have a strong opinion on this. It looks fine to me.

Reply:
Agree regarding the flash messages - perhaps I could use the value of the flash message instead of the new boolean to achieve the class toggle as well - will give a try